e2e/head_test.go

```diff
@@ -0,0 +1,407 @@
+	lastModifiedPattern := `\"[^\"]+\"`
+	typePattern := `\"[^\"]+\"`
+	sizePattern := `\d+`
+	storageClassPattern := `\"[^\"]+\"`
+
+	expectedOutput := fmt.Sprintf(
+		`{"key":"s3://%v/file.txt","etag":%s,"last_modified":%s,"type":%s,"size":%s,"storage_class":%s}`,
+		bucket,
+		etagPattern,
+		lastModifiedPattern,
+		typePattern,
+		sizePattern,
+		storageClassPattern,
+	)
+
+	match, _ := regexp.MatchString(expectedOutput, result.Combined())
+
+	if !match {
+		t.Errorf("expected output to match:\n%s\nbut got:\n%s", expectedOutput, result.Combined())
+	}
+
+	result.Assert(t, icmd.Expected{ExitCode: 0})
+}
+
+// --json head object (non-existent)
+
+func TestHeadObjectJSONNonExistent(t *testing.T) {
+	t.Parallel()
+
+	_, s5cmd := setup(t)
+
+	cmd := s5cmd("--json", "head", "s3://non-existent-bucket/non-existent-file.txt")
+	result := icmd.RunCmd(cmd)
+
+	result.Assert(t, icmd.Expected{ExitCode: 1})
+}
+
+// head object --humanize
+
+func TestHeadObjectHumanize(t *testing.T) {
+	t.Parallel()
+
+	s3client, s5cmd := setup(t)
+
+	bucket := s3BucketFromTestName(t)
+	createBucket(t, s3client, bucket)
+
+	largeFileContent := make([]byte, 1024*1024*50) // 50MB
+	for i := 0; i < len(largeFileContent); i++ {
+		largeFileContent[i] = 'a'
+	}
+	putFile(t, s3client, bucket, "file.txt", string(largeFileContent))
+
+	cmd := s5cmd("head", "--humanize", fmt.Sprintf("s3://%v/file.txt", bucket))
+	result := icmd.RunCmd(cmd)
+
+	datePattern := `\d{4}/\d{2}/\d{2} \d{2}:\d{2}:\d{2}`
+	storageClassPattern := `(?:STANDART|)`
+	sizePattern := `(?:\d+(\.\d+)?[KMGTP]?B?)`
+	s3urlPattern := `file.txt`
+
+	expectedOutput := fmt.Sprintf(
+		`%s\s+%s\s+%s\s+%s`,
+		datePattern,
+		storageClassPattern,
+		sizePattern,
+		s3urlPattern,
+	)
+
+	match, _ := regexp.MatchString(expectedOutput, result.Combined())
+
+	if !match {
+		t.Errorf("expected output to match:\n%s\nbut got:\n%s", expectedOutput, result.Combined())
+	}
+
+	result.Assert(t, icmd.Expected{ExitCode: 0})
+
+}
+
+// head object --etag
+
+func TestHeadObjectEtag(t *testing.T) {
+	t.Parallel()
+
+	s3client, s5cmd := setup(t)
+
+	bucket := s3BucketFromTestName(t)
+	createBucket(t, s3client, bucket)
+	putFile(t, s3client, bucket, "file.txt", "content")
+
+	cmd := s5cmd("head", "--etag", fmt.Sprintf("s3://%v/file.txt", bucket))
+	result := icmd.RunCmd(cmd)
+
+	//  2024/07/03 13:13:15 STANDART 9a0364b9e99bb480dd25e1f0284c8555                  7  file.txt
+
+	datePattern := `\d{4}/\d{2}/\d{2} \d{2}:\d{2}:\d{2}`
+	storageClassPattern := `(?:STANDART|)`
+	sizePattern := `\d+`
+	s3urlPattern := `file.txt`
+	etagPattern := `[a-f0-9]+`
+
+	expectedOutput := fmt.Sprintf(
+		`%s\s+%s\s+%s\s+%s\s+%s`,
+		datePattern,
+		storageClassPattern,
+		etagPattern,
+		sizePattern,
+		s3urlPattern,
+	)
+
+	match, _ := regexp.MatchString(expectedOutput, result.Combined())
+
+	if !match {
+		t.Errorf("expected output to match:\n%s\nbut got:\n%s", expectedOutput, result.Combined())
+	}
+
+	result.Assert(t, icmd.Expected{ExitCode: 0})
```

Comment:
you can use `assertLines`;

https://github.com/peak/s5cmd/blob/8355901a7be5f970a183e7d4d16d1e3b0f2ee840/e2e/cat_test.go#L156-L160

you may find more usage examples through out the e2e tests.

edit: please apply this to other tests as well.